pkgs/applications/misc/st/default.nix

```diff
@@ -4,22 +4,20 @@
 with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  name = "st-0.6";
+  name = "st";
+  version = "0.7";
   
   src = fetchurl {
-    url = "http://dl.suckless.org/st/${name}.tar.gz";
-    sha256 = "0avsfc1qp8zvshsfjwwrkvk411jlqy58z225bsdhjkl1qc40qcc5";
+    url = "http://dl.suckless.org/st/${name}-${version}.tar.gz";
+    sha1 = "358bca5bfc6ee85352dcd4f45f13f78474e5de42";
```

Comment:
Please don't downgrade to sha1, but instead keep using sha256. Even if upstream provides a weaker hash we can use a stronger hash and provide more protection to nixpkgs users. We're already [deprecating md5](https://github.com/NixOS/nixpkgs/issues/4491) and I hope sha1 goes soon, too!